client_auth.go

```diff
@@ -25,34 +25,27 @@ import (
 )
 
 type AuthRequester interface {
-	Request(server string, dataSrc AuthDataMessenger) ([]byte, error)
+	Request(client ApiClient, server string, dataSrc AuthDataMessenger) ([]byte, error)
 }
 
+// Auth client wrapper. Instantiate by yourself or use `NewAuthClient()` helper
 type AuthClient struct {
-	client *http.Client
 }
 
-func NewAuthClient(conf httpsClientConfig) (*AuthClient, error) {
-	client, err := NewHttpClient(conf)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to create auth client HTTP client")
-	}
-
-	ac := AuthClient{
-		client: client,
-	}
-	return &ac, nil
+func NewAuthClient() *AuthClient {
+	ac := AuthClient{}
+	return &ac
 }
 
-func (u *AuthClient) Request(server string, dataSrc AuthDataMessenger) ([]byte, error) {
+func (u *AuthClient) Request(client ApiClient, server string, dataSrc AuthDataMessenger) ([]byte, error) {
```

Comment:
So here `AuthClient` actually contains `http.Client` field, but still while doing request we are injecting `ApiClient`. Why we are not doing this in `NewAuthClient`? It feels that this binding makes more sense in "constructor".